src/PixiEditor/ViewModels/Nodes/NodeTypeInfo.cs

```diff
@@ -20,26 +18,42 @@ public class NodeTypeInfo
 
     public bool Hidden => PickerName is { Length: 0 };
     
+    public Type NodeViewModelType { get; }
+    
     public Type NodeType { get; }
     
     public string Icon { get; }
 
-    public NodeTypeInfo(Type type)
+    public NodeTypeInfo(Type viewModelType)
     {
-        NodeType = type;
-
-        var attribute = type.GetCustomAttribute<NodeInfoAttribute>();
+        NodeViewModelType = viewModelType;
+        NodeType = GetNodeType(NodeViewModelType.BaseType);
+        
+        var attribute = viewModelType.GetCustomAttribute<NodeViewModelAttribute>();
 
-        UniqueName = attribute.UniqueName;
         DisplayName = attribute.DisplayName;
         PickerName = attribute.PickerName;
         Category = attribute.Category ?? "";
+        Icon = attribute.Icon;
+
+        FinalPickerName = (PickerName ?? DisplayName);
+    }
 
-        if (NodeIcons.IconMap.TryGetValue(type, out var icon))
+    private Type GetNodeType(Type? baseType)
```

Comment:
This method looks _extremely_ unsafe. Argument is nullable, yet no null check is inside. Also this method will freeze if base type is not assignable to Node, sure this scenario shouldn't happen. But if you do `while (baseType != null)` it should be enough.